src/main/java/jenkinsci/plugins/influxdb/generators/SonarQubePointGenerator.java

```diff
@@ -183,19 +200,19 @@ String getSonarProjectName(String url) throws URISyntaxException {
         return projectUrl.length > 1 ? projectUrl[projectUrl.length - 1] : "";
     }
 
-    private int getLinesOfCode(String url) throws IOException {
+    private float getSonarMetric(String url, String metric) throws IOException {
         String output = getResult(url);
         JSONObject metricsObjects = JSONObject.fromObject(output);
-        int linesOfCodeCount = 0;
+        float value = 0;
```

Comment:
Have you considered not using `0` as the default value but not writing the field at all? Then it would be easier to distinguish absent data.